src/frontend/admin/src/services/api/models/Discount.ts

```diff
@@ -0,0 +1,20 @@
+export type Discount = {
+  id: string;
+  amount?: number | null;
+  rate?: number | null;
```

Comment:
`amount` and `rate` properties will always exist in the api response so they should not be optional.
```suggestion
  amount: number | null;
  rate: number | null;
```